### PR TITLE
Fixes #2376 Consolidate invalid project version messages

### LINKED ISSUE #2376: Consolidate invalid project version messages

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #2376

# Description
There are currently separate implementations of this message in PK-Sim and MoBi. They are the same, but it makes sense only to define it once.

Also, this changes the message to add the display versions to the message where possible.

## Type of change
purely caption

# Screenshots (if appropriate):
image

image

# Questions (if appropriate):

### CODE REVIEW

**Comment by msevestre** [comment]
the range is kind of weird. Where does the max comes from?

File: src/OSPSuite.Assets/UIConstants.cs [hunk 1]
```diff
@@ -1369,6 +1369,36 @@ public static class DeviationLines
...
          }
       }
+
+      private static string numberDisplay(int version) => $"(#{version})";
+
+      private static string projectNameAndVersionAsString(string versionDisplay, int version) => $"V{versionDisplay} {numberDisplay(version)}";
+
+      public static string ProjectVersionCannotBeLoaded(int projectVersion, 
+         string oldestSupportedDisplayVersion, int oldestSupportedVersion, 
+         string currentSupportedDisplayVersion,int currentSupportedVersion,
```
Comment (msevestre) [inline src/OSPSuite.Assets/UIConstants.cs:1379]: some resharper goodness required here to format

File: src/OSPSuite.Assets/UIConstants.cs [hunk 2]
```diff
@@ -1369,6 +1369,36 @@ public static class DeviationLines
...
+         }
+
+         if (projectVersion > currentSupportedVersion)
+         {
+            return $"{preamble}.\n\n" +
+                   $"The version of this project {numberDisplay(projectVersion)} is newer and cannot be loaded.\n\n" +
+                   $"Visit our download page at {downloadUrl}";
+         }
+
+         return $"Work in progress.\nThis project file is too old {numberDisplay(projectVersion)} and cannot be loaded.\nSorry :-(";
```
Comment (msevestre) [inline src/OSPSuite.Assets/UIConstants.cs:1400]: should that be gone and the 1387 to 1390 be the default?

File: src/OSPSuite.Assets/UIConstants.cs [hunk 3]
```diff
@@ -1369,6 +1369,36 @@ public static class DeviationLines
...
       }
+
+      private static string numberDisplay(int version) => $"(#{version})";
+
+      private static string projectNameAndVersionAsString(string versionDisplay, int version) => $"V{versionDisplay} {numberDisplay(version)}";
+
+      public static string ProjectVersionCannotBeLoaded(int projectVersion, 
+         string oldestSupportedDisplayVersion, int oldestSupportedVersion, 
+         string currentSupportedDisplayVersion,int currentSupportedVersion,
+         bool projectIsTooOld,
```
Comment (msevestre) [inline src/OSPSuite.Assets/UIConstants.cs:1380]: why do we need this flag ? isn it projectVersion < oldestSupportedVersion ?

**Review by msevestre — CHANGES_REQUESTED** [review]
some questions

File: src/OSPSuite.Assets/UIConstants.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/OSPSuite.Assets/UIConstants.cs:1400]: Yeah, I don't think this can ever be seen. Especially if the too old flag is removed.

**Pull request merged**